### ISSUE TITLE
Unable to get desktop notifications working.

### ISSUE THREAD

**Issue by LJNeon** [flathub/com.discordapp.Discord#473]
The flatpak is version 0.0.72, and I'm on Linux Mint 22 with Cinnamon 6.2.9 as my DE. I tried using flatseal to add org.freedesktop.Notifications to the list of allowed session busses but that didn't work. The Notifications portal says it's data has not been set up yet.

**Comment by guihkx** [comment]
Notifications should just work, no extra permissions needed.

I'm unable to verify that at the moment, but I did successfully test it [not too long ago](https://github.com/flathub/com.discordapp.Discord/issues/357#issuecomment-2389826318).

My guess is that's a configuration issue on your side...

**Comment by guihkx** [comment]
No idea... Does this issue only affect Discord among other Flatpak apps?

**Comment by guihkx** [comment]
I've set up a Mint 22 virtual machine with only Discord and Flatseal installed via Flatpak, and desktop notifications work just fine:

https://github.com/user-attachments/assets/4c8c6f8c-eb5d-46e3-84a8-746804e89534

A few other points:

- Mint is a very popular distro and Discord a very popular app, so I find it unlikely that you'd be the only one experiencing this problem at this point.
- If there is a bug somewhere (e.g. in Mint's XDG desktop portal implementation), I'm pretty sure we couldn't fix it here anyway...

For those two reasons, I'm closing this

But if you figure out a fix for this that can we can implement here, feel free to reopen this.

**Comment by guihkx** [comment]
> I was able to get the notification portal set up by installing a demo package (`flatpak install ashpd`) and sending a test notification. Notifications are now working as expected

I suppose it's possible that sending a test notification would "set up the portal data" (whatever that means), but at the same time, that only raises more questions:

- If no portal data for Notifications had been set up, how would your other Flatpak apps be sending notifications just fine before that?
- How was ASHPD able to set up the Notifications portal for Discord specifically, since it doesn't require any special permissions?

That Flatseal message about portals not being set up doesn't seem too common (at least according to my Google results), and the Flatseal dev [explained](https://github.com/tchx84/Flatseal/issues/692#issuecomment-2082759895) that those permissions are greyed out if the app hasn't requested them yet, which seems to have happened to you, but not to me.

**TL;DR:** I'm still lost here, and I don't know if there's anything actionable on our part (as a Flatpak package).

**Comment by guihkx** [comment]
>After I sent a test notification, the Notifications portal was no longer grayed out for ALL of my flatpaks not just the demo one. That leads me to believe that the portal just needed to be set up by ANY flatpak and then all could use it

That could surely explain what happened in your case, but in my case I had only Flatseal and Discord installed as Flatpak, so it'd be impossible for the notifications portal to be "set up" by any Flatpak app other than Discord...

But we'll see, I'll keep monitoring for bug reports like this one, and I'll reopen this issue if more of them show up.
